src/test/scala/org/zalando/kanadi/api/AvroBasicSpec.scala

```diff
@@ -0,0 +1,133 @@
+package org.zalando.kanadi.api
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import com.typesafe.config.ConfigFactory
+import org.specs2.Specification
+import org.specs2.concurrent.ExecutionEnv
+import org.specs2.matcher.FutureMatchers
+import org.zalando.kanadi.{Config, Utils}
+import org.zalando.kanadi.models.{EventTypeName, StreamId}
+import org.zalando.kanadi.api.EventTypeSchema.Type
+
+import java.util.UUID
+import io.circe.syntax._
+import org.mdedetrich.webmodels.FlowId
+import org.specs2.specification.{AfterAll, BeforeAll}
+import org.specs2.specification.core.SpecStructure
+import org.zalando.kanadi.api.Event.{AvroEvent, eventDecoder}
+
+import java.net.URI
+import java.util.concurrent.atomic.{AtomicBoolean, AtomicInteger}
+import scala.concurrent.{Await, Promise}
+import scala.concurrent.duration.DurationInt
+import scala.util.Success
+
+case class AvroRecordTest(foo: String, bar: Int)
+
+class AvroBasicSpec(implicit ec: ExecutionEnv)
+    extends Specification
+    with FutureMatchers
+    with Config
+    with BeforeAll
+    with AfterAll {
+
+  override val config = ConfigFactory.load()
+
+  override lazy val nakadiUri = new URI("http://localhost:8080")
+
+  override def is: SpecStructure = sequential ^
+    s2"""
+    Publish avro events $publishAvroEvent
+    Consume published avro evetns $consumeAvroEvent
+   """
+
+  implicit val system = ActorSystem()
+  implicit val http   = Http()
+
+  val eventTypeName     = EventTypeName(s"Kanadi-Test-Event-${UUID.randomUUID().toString}")
+  val OwningApplication = "KANADI"
+  val schema = """{
+    |"name": "AvroTestRecord",
+    |"namespace": "com.zalando.test",
+    |"type": "record",
+    |"fields": [
+    |{
+    | "name": "foo",
+    |  "type": "string"
+    |},
+    |{
+    |  "name": "bar",
+    |  "type": "int"
+    |}
+    |]
+    |}""".stripMargin('|')
+
+  val eventType = EventType(eventTypeName,
+                            OwningApplication,
+                            Category.Business,
+                            schema = EventTypeSchema(None, None, `type` = Type.AvroSchema, schema = schema.asJson))
+
+  val eventsTypesClient = EventTypes(nakadiUri)
+
+  val events = List(
+    AvroEvent(AvroRecordTest("foo_1", 1), Metadata(eventType = Some(eventTypeName))),
+    AvroEvent(AvroRecordTest("foo_2", 2), Metadata(eventType = Some(eventTypeName)))
+  )
+
+  val expectedDataEvents = events.map(_.data).toSet
+
+  val eventCounter                      = new AtomicInteger(0)
+  val subscriptionClosed: AtomicBoolean = new AtomicBoolean(false)
+  val streamComplete: Promise[Unit]     = Promise()
+
+  override def beforeAll = Await.result(eventsTypesClient.create(eventType), 10 seconds)
+
+  override def afterAll =
+    Await.result(eventsTypesClient.delete(eventTypeName), 10 seconds)
```

Comment:
Such tests are not sustainable and will lead to false positives depending on the environment where they run.